src/core/plugin/luapi_application.h

```diff
@@ -103,6 +104,62 @@ static int applib_saveAs(lua_State* L) {
     return args_returned;
 }
 
+/**
+ * Create a 'Open File' native dialog and return as a string
+ * the filepath the user chose to open.
+ *
+ * Examples:
+ *   path = app.getFilePath()
+ *   path = app.getFilePath({'*.bmp', '*.png'})
+ */
+static int applib_getFilePath(lua_State* L) {
+    GtkFileChooserNative* native =
+            gtk_file_chooser_native_new(_("Open file"), nullptr, GTK_FILE_CHOOSER_ACTION_OPEN, nullptr, nullptr);
+    gint res;
+    int args_returned = 1;  // change to 1 if user chooses file
```

Comment:
```suggestion
    int args_returned = 0;  // change to 1 if user chooses file
```

Reply:
Why change this?

Reply:
Each API function should return the number of return values (that are pushed onto the Lua stack). If the user chooses no file, there is no filename pushed onto the Lua stack.